packages/@aws-cdk/aws-events-targets/test/lambda/lambda.test.ts

```diff
@@ -100,7 +100,27 @@ test('adding same singleton lambda function as target mutiple times creates perm
   }));
 
   // THEN
-  expect(stack).to(countResources('AWS::Lambda::Permission', 1));
+  expect(stack).toCountResources('AWS::Lambda::Permission', 1);
+});
+
+test('lambda handler and cloudwatch event across stacks', () => {
+  // GIVEN
+  const app = new cdk.App();
+  const lambdaStack = new cdk.Stack(app, 'LambdaStack');
+
+  const fn = new lambda.Function(lambdaStack, 'MyLambda', {
+    code: new lambda.InlineCode('foo'),
+    handler: 'bar',
+    runtime: lambda.Runtime.PYTHON_2_7,
+  });
+
+  const eventStack = new cdk.Stack(app, 'EventStack');
+  new events.Rule(eventStack, 'Rule', {
+    schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+    targets: [new targets.LambdaFunction(fn)],
+  });
+
+  expect(() => app.synth()).not.toThrow();
```

Comment:
Obviously, this is testing that there's not a circular dependency, and so we don't fail. A slightly stronger assertion would verify the Function, Rule, and Permission all landed in the right stack (much like the count resources above).